test/xskbench/km/sys/xskbench_kernel.h

```diff
@@ -0,0 +1,211 @@
+//
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+//
+
+#pragma once
+
+#define printf(format, ...) \
+    PlatPrint(__FUNCTION__, __LINE__, 4, format, __VA_ARGS__)
+
+#define printf_error(format, ...) \
+    PlatPrint(__FUNCTION__, __LINE__, 3, format, __VA_ARGS__)
+
+#define printf_verbose(format, ...) \
+    if (verbose) PlatPrint(__FUNCTION__, __LINE__, 5, format, __VA_ARGS__)
+
+#define ABORT(...) \
+    printf_error(__VA_ARGS__); KeBugCheck(0xeeeeeeee)
+
+#define ASSERT_FRE(expr) \
+    if (!(expr)) { ABORT("(%s) failed line %d\n", #expr, __LINE__);}
+
+#if DBG
+#define VERIFY(expr) NT_ASSERT(expr)
+#else
+#define VERIFY(expr) (expr)
+#endif
+
+#define calloc(a, b) ExAllocatePoolZero(NonPagedPoolNx, (a) * (b), 'vDbX')
+
+#define malloc(a) ExAllocatePoolZero(NonPagedPoolNx, a, 'vDbX')
+
+#define free(a) ExFreePoolWithTag(a, 'vDbX')
+
+#define INFINITE ULONG_MAX
+
+typedef VOID (*PLAT_THREAD_FN) (VOID *Context);
+
+typedef struct _PLAT_THREAD {
+    HANDLE Thread;
+    PLAT_THREAD_FN Function;
+    VOID *Context;
+} PLAT_THREAD;
+
+// DWORD
+// WINAPI
+// PlatThreadFunctionWrapper(
+//     _In_ LPVOID lpParameter
+//     );
+
+// #ifdef DEFINE_PLAT_THREAD_FUNCTION_WRAPPER
+// inline
+// DWORD
+// WINAPI
+// PlatThreadFunctionWrapper(
+//     _In_ LPVOID lpParameter
+//     )
+// {
+//     PLAT_THREAD *Thread = (PLAT_THREAD *)lpParameter;
+//     Thread->Function(Thread->Context);
+//     return 0;
+// }
+// #endif
+
+inline
+BOOLEAN
+PlatCreateThread(PLAT_THREAD *P, PLAT_THREAD_FN F, VOID *C)
+{
+    P->Function = F;
+    P->Context = C;
+    NTSTATUS Status =
+        PsCreateSystemThread(&P->Thread, THREAD_ALL_ACCESS, NULL, NULL, NULL, F, C);
+    return Status == STATUS_SUCCESS;
+}
+
+inline
+VOID
+PlatWaitThread(PLAT_THREAD *P, ULONG TimeoutMs)
```

Comment:
Can we take a dependency on the CxPlat library used by MsQuic/usersim?

Reply:
Unfortunately, not yet. We haven't done all the refactoring to move stuff into [microsoft/cxplat](https://github.com/microsoft/cxplat) yet.

Reply:
That cxplat repository looks stale. SHould we take a dependency on https://github.com/microsoft/usersim/tree/main/cxplat ?

Reply:
@nibanks / @mtfriesen ^

Reply:
No, don't add any extra dependencies yet.

Reply:
Hmm, why not? A dependency seems better than duplicating code to me.

Reply:
Ok I have the context now that providing facilities in [cxplat](https://github.com/microsoft/cxplat) is the end goal, but it just hasn't been prioritized yet. I think we can cheaply add a cxplat copy into xdp now and migrate to the proper cxplat once it is in place. This avoids any extra work adding and then untangling a temporary dependency on usersim. 

https://github.com/microsoft/usersim/discussions/167

Reply:
This is being brought up again. cxplat copies are littered throughout microsoft github repositories (msquic, ebpf-for-windows, usersim, ntosebpfext, win-net-test). Adding another in xdp brings the count higher than can be counted on one hand. What is the tipping point here? Would it be reasonable to start populating the cxplat repo with what xdp needs and have xdp be the first consumer?

Reply:
I absolutely want to move stuff there, but just don't have the time. But I do think we've reached that tipping point.

Reply:
I can start this work by bringing in the cxplat pieces that xdp needs. Other workstreams can contribute more to cxplat as needed and eliminate cxplat copies when ready/possible. Does that seem like a reasonable plan?